### PR TITLE
DYN-4305: After panning, sometimes zooming doesn't zoom into the mouse position

### PR DESCRIPTION
**Pull request by sm6srw**
### Purpose

This pull request does:
* add code to stop sending view notifications when no changes have been made during pan/zoom 
* fix two bugs that cause the offset to not update correctly.

### Declarations

Check these if you believe they are true

- [X] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [x] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Release Notes

Initiating a zoom right after a pan no longer sporadically zooms around the wrong position.

### Reviewers

(FILL ME IN) Reviewer 1  (If possible, assign the Reviewer for the PR)

(FILL ME IN, optional) Any additional notes to reviewers or testers.

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

File: src/DynamoCoreWpf/Controls/ZoomBorder.cs [hunk 1]
```diff
@@ -100,29 +100,34 @@ public Point GetTranslateTransformOrigin()
 
         public void SetTranslateTransformOrigin(Point2D p)
         {
-            var tt = GetTranslateTransform(child);
-            tt.X = p.X;
-            tt.Y = p.Y;
+            var tt = GetChildTranslateTransform();
+            if (tt.X != p.X || tt.Y != p.Y)
```
Comment (sm6srw) [inline src/DynamoCoreWpf/Controls/ZoomBorder.cs:104]: Keeping code that stops propagation if nothing has changed.

File: src/DynamoCoreWpf/Controls/ZoomBorder.cs [hunk 2]
```diff
@@ -100,29 +100,34 @@ public Point GetTranslateTransformOrigin()
...
+                NotifyViewSettingsChanged(tt.X, tt.Y, zoom);
+            }
         }
 
         #region Child Events
 
         private void child_MouseWheel(object sender, MouseWheelEventArgs e)
         {
-            if (child != null)
+            if (child != null && !child.IsMouseCaptured)
```
Comment (sm6srw) [inline src/DynamoCoreWpf/Controls/ZoomBorder.cs:130]: Do not allow zoom if middle button is down. This caused zoom to jump.

File: src/DynamoCoreWpf/ViewModels/Core/WorkspaceViewModel.cs [hunk 3]
```diff
@@ -1310,7 +1310,7 @@ private void SetCurrentOffset(object parameter)
 
             //set the current offset without triggering
             //any property change notices.
-            if (Model.X != p.X && Model.Y != p.Y)
+            if (Model.X != p.X || Model.Y != p.Y)
```
Comment (sm6srw) [inline src/DynamoCoreWpf/ViewModels/Core/WorkspaceViewModel.cs:1313]: This was the second bug. Zoom jumped if only X or Y changed during pan.

File: src/DynamoCoreWpf/Controls/ZoomBorder.cs [hunk 4]
```diff
@@ -173,6 +178,12 @@ private void child_MouseMove(object sender, MouseEventArgs e)
 
             // Change ZoomBorder's child translation
             Vector v = start - e.GetPosition(this);
+
+            if (v.Length == 0.0)
```
Comment (pinzart90) [inline src/DynamoCoreWpf/Controls/ZoomBorder.cs:182]: maybe use a tolerance ?

File: src/DynamoCoreWpf/Controls/ZoomBorder.cs [hunk 5]
```diff
@@ -100,29 +100,34 @@ public Point GetTranslateTransformOrigin()
...
-            st.ScaleY = zoom;
-
-            var tt = GetTranslateTransform(child);
-            NotifyViewSettingsChanged(tt.X, tt.Y, zoom);
+            var st = GetChildScaleTransform();
+            if (st.ScaleX != zoom || st.ScaleY != zoom)
+            {
+                st.ScaleX = zoom;
+                st.ScaleY = zoom;
+                var tt = GetTranslateTransform(child);
```
Comment (pinzart90) [inline src/DynamoCoreWpf/Controls/ZoomBorder.cs:121]: GetChildTranslateTransform() ?
I saw you used this method at line 103

**Review by pinzart90 — APPROVED** [review]
A couple of minor comments

File: src/DynamoCoreWpf/Controls/ZoomBorder.cs (hunk 5 above)
Comment (sm6srw) [reply src/DynamoCoreWpf/Controls/ZoomBorder.cs:121]: Yes, could be used here. I may do a pass and use the child version of these methods.

File: src/DynamoCoreWpf/Controls/ZoomBorder.cs (hunk 4 above)
Comment (sm6srw) [reply src/DynamoCoreWpf/Controls/ZoomBorder.cs:182]: I was thinking of it but the values I have seen here have been 0,1,2 etc and 0 has been 0 in those cases. So I  think this is good enough.

**Pull request merged**